chore: update changelog and versioning (#445)
awslabs/aws-advanced-jdbc-wrapper: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [2.1.1] - 2023-5-15
### :bug: Fixed

> - MySQL dependency in code impacting workflows using other drivers.

Reviewer: ```suggestion
- MySQL reference in code that could impact workflows with other drivers.
```
